Add CreditCoin chain support

DBosley opened the pull request:
## Summary
Adds support for CreditCoin (Wormhole chain ID 59) to wormhole-connect.

## Changes
- Updated SDK dependencies to 3.8.5 (includes CreditCoin support)
- Added CreditCoin mainnet configuration
- Added native CTC token and wCTC (wrapped on Ethereum)
- Created chain and token icon components
- Integrated CoinGecko for native token pricing
- Updated legacy SDK configs
- Updated tests to include CreditCoin

## Chain Details
- **Wormhole Chain ID**: 59
- **Native Chain ID**: 102030
- **Native Token**: CTC (18 decimals)
- **Explorer**: Blockscout @ https://creditcoin.blockscout.com
- **RPC**: https://mainnet3.creditcoin.network

## CoinGecko Integration
⚠️ **Note**: CoinGecko does not have a Platform ID for CreditCoin in their asset_platforms API.
- ✅ Native token ID "creditcoin" added to `NATIVE_TOKEN_IDS`
- ✅ Native CTC price display works
- ❌ New tokens deployed on CreditCoin won't auto-display USD prices

## Testing
- ✅ Build: Passed
- ✅ Tests: 18/18 passing
- ✅ Lint: Passed

## Files Changed (14)
```
 package.json                       | SDK deps 3.8.1 → 3.8.5
 src/config/mainnet/chains.ts       | Chain config
 src/config/mainnet/rpcs.ts         | RPC mapping
 src/config/mainnet/tokens.ts       | CTC + wCTC tokens
 src/config/types.ts                | TokenIcon enum
 src/icons/Chains/CreditCoin.tsx    | New chain icon
 src/icons/Tokens/CTC.tsx           | New token icon
 src/icons/ChainIcons.tsx           | Icon exports
 src/icons/TokenIcons.tsx           | Icon exports
 src/utils/coingecko.ts             | CoinGecko integration
 src/sdklegacy/config/MAINNET.ts    | Legacy SDK RPC
 src/hooks/useConfigUpdate.test.tsx | Test updates
 src/routes/operator.test.ts        | Test updates
 package-lock.json                  | Dependency updates
```

DBosley commented on src/config/mainnet/tokens.ts:
```diff
@@ -62,6 +62,15 @@ export const MAINNET_TOKENS: TokenConfig[] = [
     },
   },
+  {
+    symbol: 'wCTC',
```
Not sure if we need this... I told it to add it because the Devs were talking about the differences between CTC and wCTC

priyajeet commented on src/config/mainnet/chains.ts:
```diff
@@ -233,4 +225,12 @@ export const MAINNET_CHAINS: ChainsConfig = {
     symbol: 'XRP',
   },
+  CreditCoin: {
+    displayName: 'Creditcoin',
```
camel casing

DBosley replied:
Honestly, I'm not sure what's correct.... I see both all over. I'll confirm which is correct before editing anything

DBosley replied:
The display name is correct based on their branding. The SDK was implemented with camelcasing by the WF guys. We'll need to go fix this in the SDKs if we want to fix it everywhere.

tonyjin replied:
Let's fix it everywhere now, otherwise we're just going to get asked about this later. Thanks

priyajeet commented on src/icons/ChainIcons.tsx:
```diff
@@ -64,8 +64,8 @@ const iconMap: { [key in Chain]?: React.JSX.Element } = {
   HyperEVM: HyperEVM(),
-  HyperCore: HYPERCORE(),
   XRPLEVM: XRPLEVM(),
+  CreditCoin: CreditCoin(),
```
Casing here too, I guess line 66 also. Not sure why we have it all caps for others the 1st place.

priyajeet commented on src/utils/coingecko.ts:
```diff
@@ -30,10 +30,10 @@ const NATIVE_TOKEN_IDS: Partial<Record<Chain, string>> = {
   Ink: 'ethereum',
-  HyperEVM: 'hyperliquid',
-  HyperCore: 'usd-coin',
+  CreditCoin: 'creditcoin',
```
I think it is `creditcoin-2`

tonyjin replied:
Let's use https://www.coingecko.com/en/coins/wrapped-ctc @DBosley 

After discussion with their team, they'll OTC us any necessary CTC at this price.

priyajeet commented on package.json:
rebase

DBosley committed:
feat: add CreditCoin chain support

DBosley committed:
chore: update NTT packages and fix CreditCoin icon visibility

- Bump NTT packages from 3.0.1 to 3.0.2
- Fix CreditCoin chain and token icons using currentColor for theme compatibility
